### PR TITLE
Pull request/314bc300

### PR DESCRIPTION
**Pull request by yakir12**
So sorry, for some reason the `progressbar` part didn't make it into my last PR. This one includes it.

### CODE REVIEW

File: src/widgets.jl [hunk 1]
```diff
@@ -946,3 +946,61 @@ function cyclicspinbutton{T}(range::Range{T}, carry_up::Signal{Bool};
...
+
+# convert a member of the range into a decimal 
+range2fraction(r::Range{T}, i::T) where T<:Number = (i - first(r) + step(r))/step(r)/length(r)
+
+"""
+    progressbar(range::Range; widget=nothing, signal=nothing)
+
+Create a progressbar displaying the current iteration in the given range; new iterations may be
+displayed by pushing to the widget. Note that iterators that are not members of the range are not 
+checkes for. Optionally specify
```
Comment (timholy) [inline src/widgets.jl]: checkes->checked

File: src/widgets.jl [hunk 2]
```diff
@@ -946,3 +946,61 @@ function cyclicspinbutton{T}(range::Range{T}, carry_up::Signal{Bool};
...
+        obj = new{T}(signal, widget, preserved)
+        gc_preserve(widget, obj)
+        obj
+    end
+end
+ProgressBar{T}(signal::Signal{T}, widget::GtkProgressBarLeaf, preserved) =
+    ProgressBar{T}(signal, widget, preserved)
+
+# convert a member of the range into a decimal 
+range2fraction(r::Range{T}, i::T) where T<:Number = (i - first(r) + step(r))/step(r)/length(r)
```
Comment (timholy) [inline src/widgets.jl]: How about `(i - first(r))/(last(r) - first(r))`?

File: src/widgets.jl [hunk 3]
```diff
@@ -946,3 +946,61 @@ function cyclicspinbutton{T}(range::Range{T}, carry_up::Signal{Bool};
...
+
+"""
+    progressbar(range::Range; widget=nothing, signal=nothing)
+
+Create a progressbar displaying the current iteration in the given range; new iterations may be
+displayed by pushing to the widget. Note that iterators that are not members of the range are not 
+checkes for. Optionally specify
+  - the GtkProgressBar `widget` (by default, creates a new one)
+  - the (Reactive.jl) `signal` coupled to this progressbar (by default, creates a new signal)
```
Comment (timholy) [inline src/widgets.jl]: Maybe a short usage demo?

File: src/widgets.jl (hunk 2 above)
Comment (yakir12) [reply src/widgets.jl]: Of course.

File: src/widgets.jl (hunk 1 above)
Comment (yakir12) [reply src/widgets.jl]: checked!

File: src/widgets.jl (hunk 3 above)
Comment (yakir12) [reply src/widgets.jl]: yap.

**Pull request closed without merging**